src/Modules/UngradedModules/Presentation.jl

```diff
@@ -638,3 +702,142 @@ end
+
+This is done by computing a minimal presentation.
+
+# Examples
+```jldoctest
+julia> R = ZZ;
+
+julia> F = free_module(FreeMod, R, 2);
+
+julia> A = matrix(ZZ, [2 0; 0 3]);
+
+julia> M = cokernel(hom(F, F, A));
+
+julia> is_finite(M)
+true
+
+julia> B = matrix(ZZ, [0 0; 0 0]);
+
+julia> N = cokernel(hom(F, F, B));
+
+julia> is_finite(N)
+false
+
+julia> B = matrix(ZZ, [0 0; 0 0]);
+
+julia> N = cokernel(hom(F, F, B));
+
+julia> is_finite(N)
+false
+
+julia> K, a = finite_field(7, "a");
+
+julia> G = free_module(FreeMod, K, 3);
+
+julia> C = matrix(K, [1 0 0; 0 1 0; 0 0 1]);
+
+julia> L = cokernel(hom(G, G, C));
+
+julia> is_finite(L)
+true
+
+julia> H = free_module(FreeMod, QQ, 1);
+
+julia> P = cokernel(hom(H, H, matrix(QQ, 1, 1, [QQ(0)])));
+
+julia> is_finite(P)
+false
+
+julia> Z = cokernel(hom(H, H, matrix(QQ, 1, 1, [QQ(1)])));
+
+julia> is_finite(Z)
+true
+```
+"""
+function is_finite(M::SubquoModule{T}) where {T<:Union{ZZRingElem, FieldElem}}
+  M_prime, _ = prune_with_map_atomic(M)
+  R = base_ring(M_prime)
+  pres = presentation(M_prime)
+  rel_map = map(pres, 1)
+  rel_matrix = matrix(rel_map)
+  nc = size(rel_matrix, 2)
+  nr = size(rel_matrix, 1)
+  has_free_generator = any(j -> all(i -> iszero(rel_matrix[i, j]), 1:nr), 1:nc)
+  if isa(R, ZZRing)
+    return !has_free_generator && nc > 0
+  elseif isa(R, Field)
+    if is_finite(R)
+      return true
+    else
+      return nc == 0
+    end
+  else
+    error("The base ring $(typeof(R)) is not supported.")
+  end
+end
+
+@doc raw"""
+    size(M::SubquoModule{T}) where {T<:Union{ZZRingElem, FieldElem}}
+
+Compute the cardinality of the finitely presented module `M` over `ZZRing` or a `Field` as a set.
+Returns `PosInf` if the module is infinite.
+
+This is done by computing a minimal presentation.
+
+# Examples
+```jldoctest
+julia> R = ZZ;
+
+julia> F = free_module(FreeMod, R, 2);
+
+julia> M = cokernel(hom(F, F, matrix(ZZ, [2 0; 0 3])));
+
+julia> size(M)
+6
+
+julia> N = cokernel(hom(F, F, matrix(ZZ, [1 0; 0 0])));
+
+julia> size(N)
+infinity
+
+julia> K, a = finite_field(7, "a");
+
+julia> G = free_module(FreeMod, K, 3);
+
+julia> H = free_module(FreeMod, K, 1);
+
+julia> L = cokernel(hom(H, G, matrix(K, [1 0 0])));
+
+julia> size(L)
+49
+```
+"""
+function size(M::SubquoModule{T}) where {T<:Union{ZZRingElem, FieldElem}}
+  M_prime, _ = prune_with_map_atomic(M)
+  R = base_ring(M_prime)
+  pres = presentation(M_prime)
+  rel_matrix = matrix(map(pres, 1))
+  nc, nr = size(rel_matrix, 2), size(rel_matrix, 1)
+  has_free_generator = any(j -> all(i -> iszero(rel_matrix[i, j]), 1:nr), 1:nc)
+  if isa(R, ZZRing)
+    if has_free_generator
+      return PosInf()
+    end
+    return prod(abs(rel_matrix[i,i]) for i in 1:min(nr,nc))
+  elseif isa(R, Field)
+    if is_finite(R)
+      q = order(R)
+      dim = ngens(M_prime)
+      return q^dim
+    else
+      return nc == 0 ? 1 : PosInf()
+    end
+  end
+end
```

Comment:
```suggestion
  else
    error("This should not be reached")
  end
```
... to help the Julia compiler figure out that we can not actually end up returning `nothing`.

Alternatively, change the `elseif isa(R, Field)` to `else` ...

Or split this method into two. That shouldn't be too bad...?
```suggestion
function size(M::SubquoModule{ZZRingElem})
  M_prime, _ = prune_with_map_atomic(M)
  pres = presentation(M_prime)
  rel_matrix = matrix(map(pres, 1))
  nc, nr = size(rel_matrix, 2), size(rel_matrix, 1)
  has_free_generator = any(j -> all(i -> iszero(rel_matrix[i, j]), 1:nr), 1:nc)
  if has_free_generator
    return PosInf()
  end
  return prod(abs(rel_matrix[i,i]) for i in 1:min(nr,nc))
end

function size(M::SubquoModule{T}) where {T<:FieldElem}
  M_prime, _ = prune_with_map_atomic(M)
  R = base_ring(M_prime)
  if is_finite(R)
    q = order(R)
    dim = ngens(M_prime)
    return q^dim
  else
    pres = presentation(M_prime)
    rel_matrix = matrix(map(pres, 1))
    return ncols(rel_matrix) == 0 ? 1 : PosInf()
  end
end

```